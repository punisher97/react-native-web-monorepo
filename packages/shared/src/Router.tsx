export * from 'react-router-native';
export { NativeRouter as Router } from 'react-router-native';
