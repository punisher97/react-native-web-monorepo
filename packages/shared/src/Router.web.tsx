import React from 'react';
export * from 'react-router-dom';
export { BrowserRouter as Router } from 'react-router-dom';
import { Link as BrowserLink, LinkProps } from 'react-router-dom';

export declare interface ILinkProps extends LinkProps {
  styles: React.CSSProperties;
}

export function Link({ style, ...props }: ILinkProps): JSX.Element {
  return <BrowserLink styles={style} {...props} />;
}
