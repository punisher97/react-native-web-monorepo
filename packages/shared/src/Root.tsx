import React from 'react';
import { Router, Switch, Route, useHistory } from './Router';
import { StyleSheet, Text, View, SafeAreaView, Alert, Platform } from 'react-native';
import { Backdrop, List, ListItem, Icon } from 'material-bread';

import Home from './screens/Home';
import About from './screens/About';
import NotFound from './screens/404';
import Child from './screens/Child';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 30,
    alignItems: 'center',
  },
  backdropHeader: {
    height: 56,
    flexDirection: 'row',
    alignItems: 'center',
  },
  backdropHeaderTitle: {
    fontSize: 18,
    color: 'white',
    fontWeight: '500',
    marginLeft: 72,
  },
});

const BackLayerRevealed = () => {
  const history = useHistory();
  return (
    <>
      <View style={{ flex: 1, width: '100%' }}>
        <View style={styles.backdropHeader}>
          <Text style={styles.backdropHeaderTitle}>Navigation</Text>
        </View>
        <List
          style={{
            width: '100%',
            backgroundColor: 'transparent',
          }}>
          <ListItem
            text={'Home'}
            textStyle={{ color: 'white' }}
            selected
            onPress={() => {
              history.push('/');
            }}
            style={{ backgroundColor: 'transparent' }}
            icon={<Icon name={'home'} size={24} color={'white'} />}
          />
          <ListItem
            text={'Music'}
            onPress={() => {
              history.push('/child/music');
            }}
            textStyle={{ color: 'white' }}
            style={{ backgroundColor: 'transparent' }}
            icon={<Icon name={'music-note'} size={24} color={'white'} />}
          />
          <ListItem
            text={'Favorites'}
            onPress={() => {
              history.push('/child/favorites');
            }}
            textStyle={{ color: 'white' }}
            style={{ backgroundColor: 'transparent' }}
            icon={<Icon name={'favorite'} size={24} color={'white'} />}
          />
          <ListItem
            text={'Settings'}
            onPress={() => {
              history.push('/child/settings');
            }}
            textStyle={{ color: 'white' }}
            style={{ backgroundColor: 'transparent' }}
            icon={<Icon name={'settings'} size={24} color={'white'} />}
          />
          <ListItem
            text={'About'}
            onPress={() => {
              if (Platform.OS === 'web') {
                if (confirm('You sure navigate about?')) {
                  history.push('/about');
                }
              } else {
                Alert.alert(
                  'Go to about',
                  'You sure navigate about?',
                  [
                    { text: 'Ask me later', onPress: () => console.log('Ask me later pressed') },
                    {
                      text: 'Cancel',
                      onPress: () => console.log('Cancel Pressed'),
                      style: 'cancel',
                    },
                    { text: 'OK', onPress: () => history.push('/about') },
                  ],
                  { cancelable: false },
                );
              }
            }}
            textStyle={{ color: 'white' }}
            style={{ backgroundColor: 'transparent' }}
            icon={<Icon name={'error'} size={24} color={'white'} />}
          />
        </List>
      </View>
    </>
  );
};

function Root() {
  return (
    <SafeAreaView style={{ flex: 1, backgroundColor: '#fff' }}>
      <Router>
        <View
          style={{
            flex: 1,
            width: '100%',
          }}>
          <Backdrop
            backLayerConcealed={
              <View style={styles.backdropHeader}>
                <Text style={styles.backdropHeaderTitle}>App</Text>
              </View>
            }
            backLayerRevealed={<BackLayerRevealed />}
            offset={300}>
            <View style={styles.container}>
              <Switch>
                <Route exact path="/" component={Home} />
                <Route path="/about" component={About} />
                <Route path="/child/:id" component={Child} />
                {Platform.OS === 'web' && <Route component={NotFound} />}
              </Switch>
            </View>
          </Backdrop>
        </View>
      </Router>
    </SafeAreaView>
  );
}

export default Root;
