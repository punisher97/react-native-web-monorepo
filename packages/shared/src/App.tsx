import React from 'react';
import { BreadProvider } from 'material-bread';
import Root from './Root';

export function App() {
  return (
    <BreadProvider>
      <Root />
    </BreadProvider>
  );
}

declare var global: any;
