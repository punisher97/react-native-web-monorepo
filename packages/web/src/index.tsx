import { AppRegistry } from 'react-native';
import './styles/global.css';
import { App } from 'shared/src/App';

AppRegistry.registerComponent('myprojectname', () => App);
AppRegistry.runApplication('myprojectname', {
  rootTag: document.getElementById('root'),
});
