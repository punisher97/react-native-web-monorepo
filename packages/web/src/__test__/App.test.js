/**
 * @format
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, waitForElement } from '@testing-library/react';
import { App } from 'shared/src/App';
import Root from 'shared/src/Root';

// Note: test renderer must be required after react-native.

test('renders correctly App', () => {
  render(<App />);
});

test('Renders the shared root', async () => {
  const { getByText, baseElement } = render(<Root />);
  const theText = 'App';
  await waitForElement(() => getByText(theText));
  expect(getByText(theText)).toHaveTextContent(theText);
  expect(baseElement).toMatchSnapshot();
});
