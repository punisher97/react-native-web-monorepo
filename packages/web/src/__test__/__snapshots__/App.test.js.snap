// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the shared root 1`] = `
<body>
  <div>
    <div
      class="css-view-1dbjc4n"
      style="background-color: rgb(255, 255, 255); flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
    >
      <div
        class="css-view-1dbjc4n"
        style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%; width: 100%;"
      >
        <div
          class="css-view-1dbjc4n"
          style="background-color: rgb(30, 136, 229); flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
        >
          <div
            class="css-view-1dbjc4n"
            style="bottom: 0px; opacity: 1; position: absolute; top: 0px; width: 100%;"
          >
            <div
              class="css-view-1dbjc4n"
              style="align-items: center; flex-direction: row; height: 56px;"
            >
              <div
                class="css-text-901oao"
                dir="auto"
                style="color: rgb(255, 255, 255); font-size: 18px; font-weight: 500; margin-left: 72px;"
              >
                App
              </div>
            </div>
          </div>
          <div
            class="css-view-1dbjc4n"
            style="background-color: rgb(255, 255, 255); border-top-left-radius: 16px; border-top-right-radius: 16px; flex-grow: 1; flex-shrink: 1; flex-basis: 0%; transform: translateY(56px);"
          >
            <div
              class="css-text-901oao"
              dir="auto"
              style="color: rgba(0, 0, 0, 0.8); font-size: 16px; margin-bottom: 0px; margin-left: 20px; margin-top: 16px; text-align: left;"
            />
            <div
              class="css-view-1dbjc4n"
              style="align-items: center; flex-grow: 1; flex-shrink: 1; flex-basis: 0%; padding-top: 30px;"
            >
              <div
                class="css-text-901oao"
                dir="auto"
                style="align-self: flex-start; color: rgba(0, 0, 0, 0.87); font-size: 20px; margin-bottom: 0px; margin-left: 20px; text-align: left;"
                text="Albums"
              >
                Albums
              </div>
              <div
                class="css-view-1dbjc4n"
                data-focusable="true"
                data-testid="mb-ripple"
                style="align-items: center; background-color: rgba(0, 0, 0, 0); cursor: pointer; flex-direction: row; padding: 12px 16px 12px 16px; width: 100%; z-index: 1;"
                tabindex="0"
              >
                <div
                  class="css-view-1dbjc4n"
                  style="align-items: center; background-color: rgb(163, 163, 163); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; height: 40px; justify-content: center; width: 40px;"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(236, 236, 236); font-family: MaterialIcons; font-size: 26.666666666666668px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="align-self: center; flex-shrink: 1; margin-left: 16px;"
                >
                  <div
                    class="css-view-1dbjc4n"
                    style="align-items: flex-start;"
                  >
                    <div
                      class="css-text-901oao css-textOneLine-bfa6kz"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.87); font-size: 16px; letter-spacing: 0.5px; line-height: 16px; margin-bottom: 0px; text-align: left;"
                    >
                      Back in Black
                    </div>
                    <div
                      class="css-text-901oao css-textMultiLine-cens5h"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.57); font-size: 14px; letter-spacing: 0.4px; line-height: 14px; margin-bottom: 0px; margin-top: 4px; text-align: left;"
                    >
                      AC/DC
                    </div>
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
                />
                <div
                  class="css-view-1dbjc4n"
                  data-focusable="true"
                  data-testid="mb-ripple"
                  style="align-items: center; cursor: pointer; height: 32px; justify-content: center; width: 32px;"
                  tabindex="0"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(110, 110, 110); font-family: MaterialIcons; font-size: 24px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                  <div
                    class="css-view-1dbjc4n r-pointerEvents-633pao"
                    style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 100px; border-top-right-radius: 100px; border-bottom-right-radius: 100px; border-bottom-left-radius: 100px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                  />
                </div>
                <div
                  class="css-view-1dbjc4n r-pointerEvents-633pao"
                  style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                />
              </div>
              <div
                class="css-view-1dbjc4n"
                data-focusable="true"
                data-testid="mb-ripple"
                style="align-items: center; background-color: rgba(0, 0, 0, 0); cursor: pointer; flex-direction: row; padding: 12px 16px 12px 16px; width: 100%; z-index: 1;"
                tabindex="0"
              >
                <div
                  class="css-view-1dbjc4n"
                  style="align-items: center; background-color: rgb(163, 163, 163); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; height: 40px; justify-content: center; width: 40px;"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(236, 236, 236); font-family: MaterialIcons; font-size: 26.666666666666668px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="align-self: center; flex-shrink: 1; margin-left: 16px;"
                >
                  <div
                    class="css-view-1dbjc4n"
                    style="align-items: flex-start;"
                  >
                    <div
                      class="css-text-901oao css-textOneLine-bfa6kz"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.87); font-size: 16px; letter-spacing: 0.5px; line-height: 16px; margin-bottom: 0px; text-align: left;"
                    >
                      Hotel California
                    </div>
                    <div
                      class="css-text-901oao css-textMultiLine-cens5h"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.57); font-size: 14px; letter-spacing: 0.4px; line-height: 14px; margin-bottom: 0px; margin-top: 4px; text-align: left;"
                    >
                      Eagles
                    </div>
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
                />
                <div
                  class="css-view-1dbjc4n"
                  data-focusable="true"
                  data-testid="mb-ripple"
                  style="align-items: center; cursor: pointer; height: 32px; justify-content: center; width: 32px;"
                  tabindex="0"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(110, 110, 110); font-family: MaterialIcons; font-size: 24px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                  <div
                    class="css-view-1dbjc4n r-pointerEvents-633pao"
                    style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 100px; border-top-right-radius: 100px; border-bottom-right-radius: 100px; border-bottom-left-radius: 100px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                  />
                </div>
                <div
                  class="css-view-1dbjc4n r-pointerEvents-633pao"
                  style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                />
              </div>
              <div
                class="css-view-1dbjc4n"
                data-focusable="true"
                data-testid="mb-ripple"
                style="align-items: center; background-color: rgba(0, 0, 0, 0); cursor: pointer; flex-direction: row; padding: 12px 16px 12px 16px; width: 100%; z-index: 1;"
                tabindex="0"
              >
                <div
                  class="css-view-1dbjc4n"
                  style="align-items: center; background-color: rgb(163, 163, 163); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; height: 40px; justify-content: center; width: 40px;"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(236, 236, 236); font-family: MaterialIcons; font-size: 26.666666666666668px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="align-self: center; flex-shrink: 1; margin-left: 16px;"
                >
                  <div
                    class="css-view-1dbjc4n"
                    style="align-items: flex-start;"
                  >
                    <div
                      class="css-text-901oao css-textOneLine-bfa6kz"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.87); font-size: 16px; letter-spacing: 0.5px; line-height: 16px; margin-bottom: 0px; text-align: left;"
                    >
                      Dark Side of the Moon
                    </div>
                    <div
                      class="css-text-901oao css-textMultiLine-cens5h"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.57); font-size: 14px; letter-spacing: 0.4px; line-height: 14px; margin-bottom: 0px; margin-top: 4px; text-align: left;"
                    >
                      Pink Floyd 
                    </div>
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
                />
                <div
                  class="css-view-1dbjc4n"
                  data-focusable="true"
                  data-testid="mb-ripple"
                  style="align-items: center; cursor: pointer; height: 32px; justify-content: center; width: 32px;"
                  tabindex="0"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(110, 110, 110); font-family: MaterialIcons; font-size: 24px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                  <div
                    class="css-view-1dbjc4n r-pointerEvents-633pao"
                    style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 100px; border-top-right-radius: 100px; border-bottom-right-radius: 100px; border-bottom-left-radius: 100px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                  />
                </div>
                <div
                  class="css-view-1dbjc4n r-pointerEvents-633pao"
                  style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                />
              </div>
              <div
                class="css-view-1dbjc4n"
                data-focusable="true"
                data-testid="mb-ripple"
                style="align-items: center; background-color: rgba(0, 0, 0, 0); cursor: pointer; flex-direction: row; padding: 12px 16px 12px 16px; width: 100%; z-index: 1;"
                tabindex="0"
              >
                <div
                  class="css-view-1dbjc4n"
                  style="align-items: center; background-color: rgb(163, 163, 163); border-top-left-radius: 20px; border-top-right-radius: 20px; border-bottom-right-radius: 20px; border-bottom-left-radius: 20px; height: 40px; justify-content: center; width: 40px;"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(236, 236, 236); font-family: MaterialIcons; font-size: 26.666666666666668px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="align-self: center; flex-shrink: 1; margin-left: 16px;"
                >
                  <div
                    class="css-view-1dbjc4n"
                    style="align-items: flex-start;"
                  >
                    <div
                      class="css-text-901oao css-textOneLine-bfa6kz"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.87); font-size: 16px; letter-spacing: 0.5px; line-height: 16px; margin-bottom: 0px; text-align: left;"
                    >
                      Led Zeppelin IV
                    </div>
                    <div
                      class="css-text-901oao css-textMultiLine-cens5h"
                      dir="auto"
                      style="color: rgba(0, 0, 0, 0.57); font-size: 14px; letter-spacing: 0.4px; line-height: 14px; margin-bottom: 0px; margin-top: 4px; text-align: left;"
                    >
                      Led Zeppelin
                    </div>
                  </div>
                </div>
                <div
                  class="css-view-1dbjc4n"
                  style="flex-grow: 1; flex-shrink: 1; flex-basis: 0%;"
                />
                <div
                  class="css-view-1dbjc4n"
                  data-focusable="true"
                  data-testid="mb-ripple"
                  style="align-items: center; cursor: pointer; height: 32px; justify-content: center; width: 32px;"
                  tabindex="0"
                >
                  <div
                    class="css-text-901oao r-pointerEvents-633pao"
                    dir="auto"
                    style="background-color: rgba(0, 0, 0, 0); color: rgb(110, 110, 110); font-family: MaterialIcons; font-size: 24px; font-style: normal; font-weight: normal;"
                    theme="[object Object]"
                  >
                    
                  </div>
                  <div
                    class="css-view-1dbjc4n r-pointerEvents-633pao"
                    style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 100px; border-top-right-radius: 100px; border-bottom-right-radius: 100px; border-bottom-left-radius: 100px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                  />
                </div>
                <div
                  class="css-view-1dbjc4n r-pointerEvents-633pao"
                  style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-view-1dbjc4n"
            style="align-items: center; height: 56px; justify-content: center; position: absolute; transform: rotate(180deg); width: 56px;"
          >
            <div
              class="css-view-1dbjc4n"
              data-focusable="true"
              data-testid="mb-ripple"
              style="cursor: pointer; height: 32px; justify-content: center; width: 32px;"
              tabindex="0"
            >
              <div
                class="css-text-901oao r-pointerEvents-633pao"
                dir="auto"
                style="align-self: center; background-color: rgba(0, 0, 0, 0); color: rgb(255, 255, 255); font-family: MaterialIcons; font-size: 24px; font-style: normal; font-weight: normal;"
                theme="[object Object]"
              >
                
              </div>
              <div
                class="css-view-1dbjc4n r-pointerEvents-633pao"
                style="background-color: rgba(0, 0, 0, 0); border-top-left-radius: 100px; border-top-right-radius: 100px; border-bottom-right-radius: 100px; border-bottom-left-radius: 100px; bottom: 0px; left: 0px; overflow-x: hidden; overflow-y: hidden; position: absolute; right: 0px; top: 0px; z-index: 10;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
